### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by chriswhilar**

### CODE REVIEW

**Review by mariusu-msft — COMMENTED** [review]
Update PR title to indicate what you are adding please

File: samples/Add Voice Calling/Web/JavaScript/public/index.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿<!-- © Microsoft Corporation. All rights reserved. -->
+
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="utf-8" />
+  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/public/index.html:8]: we shouldn't be loading assets from 3p domains, 
I assume this is for UI - should we switch to Fluent UI ?

File: samples/Add Voice Calling/Web/JavaScript/src/App.css [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+.card {
+  background-color: #201f1e;
+  border: 1px solid #605e5c;
+  border-radius: 0px;
+  padding: 3em;
+  margin-top: 3em;
+}
+
+.participants-panel {
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/App.css:50]: empty rules , remove all in this file

File: samples/Add Voice Calling/Web/JavaScript/src/App.css [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+.primary-button:disabled, .primary-button:disabled {
+  background-color: #605e5c;
+  border-color: #605e5c;
+  outline: none;
+  color: black;
+}
+
+.secondary-button {
+  color: black;
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/App.css:107]: #000 and everywhere else

File: samples/Add Voice Calling/Web/JavaScript/src/App.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+function App() {
+  return (
+    <div className="App">
+      <div className="header row justify-content-between">
+        <h4>
+          Azure Communication Services - Calling SDK for Javascript
+        </h4>
+        <p>
+          More documentation on how to use the ACS Calling SDK for Javascript can be found on <a className="sdk-docs-link" target="_blank" href="https://review.docs.microsoft.com/en-us/azure/project-spool/quickstarts/voice-video-calling/calling-client-samples?branch=pr-en-us-104477&pivots=platform-web">Microsoft Docs</a>
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/App.js:17]: update this link and point to new official docs

File: samples/Add Voice Calling/Web/JavaScript/src/App.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+// © Microsoft Corporation. All rights reserved.
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/App.js:1]: if sample is MIT - remove these headers

File: samples/Add Voice Calling/Web/JavaScript/src/App.test.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+// © Microsoft Corporation. All rights reserved.
+
+import React from 'react';
+import { render } from '@testing-library/react';
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/App.test.js:4]: testing-library ?

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/AddParticipantPopover.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
+// © Microsoft Corporation. All rights reserved.
+
+import React, { useState, useRef } from "react";
+import { OverlayTrigger, Popover } from "react-bootstrap";
+import { Button, TextField } from 'office-ui-fabric-react'
+import { CommunicationUser } from '@azure/communication-calling';
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/AddParticipantPopover.js:6]: I don't think we're exporting CommunicationUser from our bundle anymore

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/AddParticipantPopover.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+export default function AddParticipantPopover(props) {
+    const [userId, setUserId] = useState('');
+
+    function handleAddCommunicationUser() {
+        console.log('handleAddCommunicationUser', userId);
+        try {
+            props.call.addParticipant({ communicationUserId: userId });
+            window.document.body.click();
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/AddParticipantPopover.js:15]: this is hacky? what's the issue here? shouldn't need try catch etc.. and that fake click is not needed?

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/AddParticipantPopover.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        } catch (e) {
+            console.error(e);
+        }
+    }
+
+    function handleAddPhoneNumber() {
+        console.log('handleAddPhoneNumber', userId);
+        try {
+            props.call.addParticipant({ phoneNumber: userId });
+            window.document.body.click();
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/AddParticipantPopover.js:25]: same as above

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/CallEndReasonCard.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+// © Microsoft Corporation. All rights reserved.
+import React from "react";
+export default function CallEndReasonCard(props) {
+    const callEndReason = props.callEndReason;
+    return (
+        <div><span>Previous call end reason</span>
+            <ul>
+                <li><span style={{ color: callEndReason.message !== 'Success' ? 'red' : ''}}>Message: {callEndReason.message}</span></li>
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/CallEndReasonCard.js:8]: red = #ff0000

Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/CallEndReasonCard.js:8]: we're not exposing message anymore

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+    provisionNewUser = async () => {
+        try {
+            this.setState({ showSpinner: true});
+            this.userDetails = await utils.provisionNewUser();
+            this.setState({ id: utils.getIdentifierText(this.userDetails.user) });
+            await this.props.onLoggedIn({ id: this.state.id, token: this.userDetails.token });
+            this.setState({ loggedIn: true });
+        } catch (Error) {
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js:26]: nit, lower case Error, "Error" is a JS keyword

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+        response.json(tokenResponse);
+    } catch(error) {
+        console.log(error);
+    }
+});
+
+/********************************************************************************************************
+ *   Client application initializing the ACS Calling Client Web SDK after receiving token from service   *
+ *********************************************************************************************************/
+import {
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js:54]: these imports make sense in TS , not in JS

Comment (akania) [reply samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js:54]: same below, e.g. CallClientOptions is a TS type, all these things will be `undefined` when imported in JS

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+});
+
+/********************************************************************************************************
+ *   Client application initializing the ACS Calling Client Web SDK after receiving token from service   *
+ *********************************************************************************************************/
+import { 
+    CallClient, CallAgent, Call, CallClientOptions,
+    RemoteParticipant, RemoteVideoStream, RemoteVideoRenderer, RemoteParticipantState,
+} from '@azure/communication-calling';
+import { CommunicationUserCredentials } from '@azure/communication-common';
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js:58]: here and everywhere else, common now exports AzureCommunicationUserCredential class , CommunicationUserCredentials  is a type

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+import { createClientLogger, setLogLevel } from '@azure/logger';
+
+export class MyCallingApp {
+    private callClient: CallClient;
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js:62]: again, this is TS , let's not mix TS into this JS sample, or we turn it into TS

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+    private callAgent: CallAgent;
+    private deviceManager: DeviceManager;
+    private currentCall: Call;
+
+    public async initCallClient() {
+        const response = (await fetch('/tokens/provisionUser')).json();
+        const token = response.token;
+        const tokenCredential = new AzureCommunicationUserCredential(token);
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js:70]: please try out this code out - in this snippet we don't import `AzureCommunicationUserCredential` yet somehow we're using it

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+
+        // Create Azure logger
+        const logger = createClientLogger('ACS');
+        setLogLevel('verbose');
+        // Redirect logger output to wherever desired
+        logger.verbose.log = (...args) => { console.log(...args); };
+        logger.info.log = (...args) => { console.info(...args) ; };
+        logger.warning.log = (...args) => { console.warn(...args); };
+        logger.error.log = (...args) => { console.error(...args); };
+        const options = {logger: logger};
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js:80]: nit
const options = { logger };

File: samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js [hunk 17]
```diff
@@ -0,0 +1,175 @@
...
+                        addedParticipant.on('participantStateChanged', participantStateChangedHandler);
+
+                        // Subscribe to is muted changed event.
+                        addedParticipant.on('isMutedChanged', isMutedChangedHandler);
+
+                        // Subscribe to video streams updated event.
+                        addedParticipant.on('videoStreamsUpdated', handleStreamsUpdated);
+
+                        //Subscribe to screen sharing streams updated event
+                        addedParticipant.on('screenSharingStreamsUpdated', handleStreamsUpdated);
```
Comment (akania) [inline samples/Add Voice Calling/Web/JavaScript/src/MakeCall/Login.js:106]: this is now obsolete - we have only videoStreams ( with type )

**Pull request closed without merging**